mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts

```diff
@@ -0,0 +1,60 @@
+import type {
+  JoinableTable,
+  JoinableTablesResult,
+} from '@mathesar/api/tables/joinable_tables';
+import { MissingExhaustiveConditionError } from '@mathesar/utils/errors';
+
+export type TableLinkType = 'in_this_table' | 'from_other_tables';
+export type TableLink = {
+  table: { id: number; name: string };
+  column: { id: number; name: string };
+};
+
+function generateTableLinkFromJoinableTable(
+  linkedTable: JoinableTable,
+  joinableTablesResult: JoinableTablesResult,
+): TableLink {
+  const isLinkReversed = linkedTable.fk_path[0][1];
+  const columnId = isLinkReversed
+    ? linkedTable.jp_path[0][1]
+    : linkedTable.jp_path[0][0];
+  console.log({ columnId });
+  return {
+    table: {
+      id: linkedTable.target,
+      name: joinableTablesResult.tables[linkedTable.target].name,
+    },
+    column: {
+      id: columnId,
+      name: joinableTablesResult.columns[columnId].name,
```

Comment:
Always assume that these properties could be undefined.

i.e. `joinableTablesResult.tables[linkedTable.target]` and `joinableTablesResult.columns[columnId]`.

The API currently does not provide the column & table details of the base table that is part of the request. As a result this will always result in an error when there are FK links to other tables in the base table. Was this code tested thoroughly?

Reply:
>The API currently does not provide the column & table details of the base table that is part of the request. As a result this will always result in an error when there are FK links to other tables in the base table. Please test the code and check API responses after implementing the logic.

@pavish do we have any documentation for this or any other APIs for that matter?

Reply:
We do not have any comprehensive documentation. We use the browsable API view provided by Django to get the request/response structure which is often sufficient. If you need help with understanding a API better, it's quicker to get in touch with the backend end.

Reply:
9e47548c